src/main/java/com/example/demo/UserRepository.java

```diff
@@ -1,35 +1,50 @@
 package com.example.demo;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
 import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
 import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
 import org.springframework.stereotype.Repository;
 
 import javax.sql.DataSource;
 import java.util.List;
 import java.util.Optional;
 
+
 @Repository
 public class UserRepository {
 
     private final Logger logger = LoggerFactory.getLogger(UserRepository.class);
-    private final ObjectMapper objectMapper;
 
     private final JdbcTemplate template;
     private final NamedParameterJdbcTemplate namedTemplate;
 
     @Autowired
-    public UserRepository(ObjectMapper objectMapper, DataSource dataSource) {
-        this.objectMapper = objectMapper;
+    public UserRepository(DataSource dataSource) {
         this.template = new JdbcTemplate(dataSource);
         this.namedTemplate = new NamedParameterJdbcTemplate(dataSource);
     }
 
     public List<User> getAllUsers() {
-        return null;
+
+        String query = """
+                SELECT id, first_name, last_name, email FROM users
+                                            WHERE email LIKE :emailFilter
+                """;
+
+        MapSqlParameterSource paramMap = new MapSqlParameterSource() {{
```

Comment:
Is this paramMap section requisite when using 'LIKE'?  I hadn't encountered the syntax, "WHERE email LIKE **:emailFilter**" that uses a colon like that.
Could you have removed the paramMap for emailFilter and just had the SQL look like, "WHERE email LIKE '%@example.com'"?
I am assuming though that this method of filtering that you used is common practice. I just like understanding what is required vs. what is best-practice.

Reply:
Using ParamMap is crucial best practice to protect from SQL injection. SQL injection is a terrible vulnerability when taking input from clients, and then injecting them into the SQL. If done without a parammap, it would give hackers the ability to gain access to data they should not have, including system information. 

The use of ParamMap is a way to prevent this as the namedtemplate takes the map, and escapes inputs passed to the SQL statement. This prevent clients from passing strings with backticks and the like to try and get more information.